.github/workflows/ci.yaml

```diff
@@ -224,6 +224,37 @@ jobs:
         with:
           token: ${{ secrets.CODECOV_TOKEN }}
 
+  mutation-test:
+    name: Mutation Testing
+
+    if: github.event_name == 'pull_request'
+
+    needs:
+      - end-to-end-tests
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Install PHP with extensions
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: 8.3
+          coverage: pcov
+          extensions: none, ctype, curl, date, dom, json, libxml, mbstring, phar, simplexml, soap, tokenizer, xml, xmlwriter, zlib
+          ini-values: assert.exception=1, zend.assertions=1, error_reporting=-1, log_errors_max_len=0, display_errors=On
+          tools: none
+
+      - name: Install dependencies with Composer
+        run: ./tools/composer install --no-ansi --no-interaction --no-progress
+
+      - name: Run mutation testing
+        run: |
+          git fetch origin $GITHUB_BASE_REF
+          php tools/infection.phar --threads=max --map-source-class-to-test --git-diff-lines --git-diff-base=origin/$GITHUB_BASE_REF --ignore-msi-with-no-mutations --only-covered          
```

Comment:
I think we can't use `--map-source-class-to-test` for this repository, as we don't have here 1-to-1 relation between test file name and source class file name.

With this option, Infection tries to find test file just by adding `*Test` postfix to the source class file name, but it won't work here from what I see because 

- tests can have multiple `CoversClass()` attributes
- tests can have no `CoversClass()` attribute
- `CoversClass()` can have source class with the name not matching naming strategy where we just remove `*Test` postfix to get the source class file name

Thus, it will lead to much less tests being executed for particular changed line than needed, which will lead to more escaped mutants.

Example:

https://github.com/sebastianbergmann/phpunit/blob/8d485f1673ee40e921517116a283ad36a5a7260d/tests/unit/Runner/Filter/TestIdFilterIteratorTest.php#L21-L24

Here we have 2 `CoversClass()`, but with current `--map-source-class-to-test` implementation Infection would run this test only when `TestIdFilterIterator` is mutated, because test file name is named just by adding `*Test` postfix. But it would not run this test file if `TestSuiteIterator` is mutated, because naming strategy is not the same. As I said above, we need to implement this feature at Infection to "understand" `CoversClass()` in a more advanced way. That's why I recommend to remove `--map-source-class-to-test` so all the unit tests will be executed and Infection will understand which tests cover changed lines from coverage report.

A bit of theory here: https://infection.github.io/guide/command-line-options.html#map-source-class-to-test

we can (should) add another strategy at Infection level to find test files by analyzing their `CoversClass(...)` attributes, but this is yet to be done. Let's handle this in the future as separate PRs.

Reply:
> we can (should) add another strategy at Infection level to find test files by analyzing their CoversClass(...) attributes

Good idea!